packages-next/auth/src/index.ts

```diff
@@ -339,6 +334,17 @@ export function createAuth<GeneratedListTypes extends BaseGeneratedListTypes>({
           return (await adminPageMiddleware(args)) ?? keystoneConfig?.ui?.pageMiddleware?.(args);
         },
         enableSessionItem: true,
+        isAccessAllowed: async (context: KeystoneContext) => {
+          // Allow access to the adminMeta data from the /init path to correctly render that page
+          // even if the user isn't logged in (which should always be the case if they're seeing /init)
+          const headers = context.req?.headers;
+          const url = headers?.referer ? new URL(headers.referer) : undefined;
+          return (
+            (url?.pathname === '/init' && url?.host === headers?.host) ||
+            keystoneConfig.ui?.isAccessAllowed?.(context) ||
+            false
```

Comment:
This should be like this:

https://github.com/keystonejs/keystone/blob/481e456ac4158207436ddd9be18fdca0f27b6409/packages-next/admin-ui/src/system/createAdminUIServer.ts#L29-L31